src/map.rs

```diff
@@ -575,6 +659,99 @@ impl MapData {
             view_box,
         )))
     }
+
+    fn parse_map_info(&mut self, info: MapV2Info) {
+        let parse_coords = |s: &str| -> Option<(f32, f32)> {
+            let mut it = s.splitn(2, ',');
+            let x = it.next()?.parse::<f32>().ok()?;
+            let y = it.next()?.parse::<f32>().ok()?;
+            Some((x, y))
+        };
+
+        let mut outlines = Vec::new();
+        let mut areas = Vec::new();
+        let mut block_lines = Vec::new();
+
+        for group in &info {
+            let Some(first) = group.first() else { continue };
+            match first.as_str() {
+                "1" => {
+                    for entry in group.iter().skip(1).filter(|e| !e.is_empty()) {
+                        let parts = entry.split(';').filter(|s| !s.is_empty()).skip(1);
+                        let mut path_points = Vec::new();
+
+                        for spec in parts {
+                            let mut coords = spec.splitn(3, ',');
+                            if let (Some(x_str), Some(y_str)) = (coords.next(), coords.next()) {
+                                if let (Ok(x), Ok(y)) = (x_str.parse::<f32>(), y_str.parse::<f32>())
+                                {
+                                    let mut p = calc_point(x, y);
+                                    p.connected = coords.next().unwrap_or("1").trim() != "3-1-0";
+                                    path_points.push(p);
+                                }
+                            }
+                        }
+
+                        if let Some(first) = path_points.first().filter(|p| p.connected) {
+                            path_points.push(Point {
+                                x: first.x,
+                                y: first.y,
+                                connected: true,
+                            });
+                        }
+                        outlines.push(path_points);
+                    }
+                }
+                "2" | "6" => {
+                    let dest = if group[0] == "2" {
+                        &mut areas
+                    } else {
+                        &mut block_lines
+                    };
+
+                    for entry in group.iter().skip(1).filter(|e| !e.is_empty()) {
+                        let poly_points: Vec<Point> = entry
+                            .split(';')
+                            .filter(|s| !s.is_empty())
+                            .skip(1)
+                            .filter_map(parse_coords)
+                            .map(|(x, y)| calc_point(x, y))
+                            .collect();
+
+                        if poly_points.len() >= 3 {
+                            dest.push(poly_points);
+                        }
+                    }
```

Comment:
Move it into a separate function, which should return dest so we can assign it to the correct value instead during the magic above